src/shaders/_prelude_lighting.glsl

```diff
@@ -0,0 +1,16 @@
+#ifdef LIGHTING_3D_MODE
```

Comment:
nit: I used `UNIFORM_LIGHTING` in https://github.com/mapbox/mapbox-gl-js/pull/12373 
what do you think if we use:

`UNIFORM_LIGHTING` for fill extrusions
`UNIFORM_LIGHTING_2D_LAYERS` for 2d layers

or 
`LIGHTING_3D_MODE` for FE
`LIGHTING_3D_MODE_2D_LAYERS` for 2d layers

Of course at the end of the day it doesn't make a difference except the naming is bit more coherent.

Reply:
No need to have separate defines for different shaders. The feature is either used or not.

Reply:
OK in this case I assume that `_prelude_lighting.glsl` is meant to be used only with 2d layers.

Reply:
No. Both the file and the preprocessor define should be used to conditionally run lighting code that's exclusive to the lighting model